static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,90 @@
+define([
+    'common/utils/fastdom-promise',
+    'Promise',
+    'common/utils/$',
+    'common/utils/mediator',
+    'common/utils/detect'
+], function (
+    fastdom,
+    Promise,
+    $,
+    mediator,
+    detect) {
+
+    return function () {
+        if (detect.getBreakpoint() !== 'mobile') {
+            fastdom.read(function () {
```

Comment:
do promises run on a background thread?  I was under the impression they still used the main thread